patroni/postgresql/slots.py

```diff
@@ -48,6 +48,20 @@ def sync_replication_slots(self, cluster):
 
                 # drop old replication slots which are not presented in desired slots
                 for name in set(self._replication_slots) - set(slots):
+                    ignore_slot = False
+                    for ignore_slots_matcher in cluster.config.ignore_slots_matchers:
+                        slot = self._replication_slots[name]
+                        if (
+                                (ignore_slots_matcher.get("name") is None or ignore_slots_matcher["name"] == name)
+                                and (ignore_slots_matcher.get("database") is None or ignore_slots_matcher["database"] == slot["database"])
+                                and (ignore_slots_matcher.get("plugin") is None or ignore_slots_matcher["plugin"] == slot["plugin"])
+                                and (ignore_slots_matcher.get("type") is None or ignore_slots_matcher["type"] == slot["type"])
+                            ):
+                            ignore_slot = True
+                            break
+                    if ignore_slot:
+                        break
```

Comment:
I think we should not break the outer loop, there might be some slots which must be deleted.
I.e. we need something like:
```python
if not ignore_slot and not self.drop_replication_slot(name):
...
```
```

Reply:
Fixing.